corehq/apps/app_manager/views/releases.py

```diff
@@ -290,11 +291,10 @@ def save_copy(request, domain, app_id):
             timer = datadog_bucket_timer('commcare.app_build.new_release', tags=[],
                                          timing_buckets=(1, 10, 30, 60, 120, 240))
             with timer:
-                copy = app.make_build(
-                    comment=comment,
-                    user_id=user_id,
-                )
-                copy.save(increment_version=False)
+                result = make_app_build(app, comment, user_id)
+                if isinstance(result, HttpResponse):
+                    return result
+                copy = result
             CouchUser.get(user_id).set_has_built_app()
         finally:
             # To make a RemoteApp always available for building
```

Comment:
i am confused about this, if anyone has context please respond.
i could not track back why this was added.
[This](https://docs.python.org/3/tutorial/errors.html#defining-clean-up-actions) tells me that the finally block will get executed even in case of a `return` which i have added now.
But even before this change, if for a linked app, the build fails, it's version is still incremented? It's not clear to me if thats harmful(because we currently use version to compare versions between master and linked app) but also why is this being done?

Reply:
I don't know exactly what this block is for, but `RemoteApp` is for the old-style apps where you edited the build files by hand on github, not remote linked apps. Remotes apps are effectively dead but the code hasn't yet been cleaned up.